pkg/client/cache/store.go

```
@@ -78,38 +114,39 @@ func (c *cache) List() []interface{} {
 	return list
 }
 
-// ContainedIDs returns a util.StringSet containing all IDs of the stored items.
-// This is a snapshot of a moment in time, and one should keep in mind that
-// other go routines can add or remove items after you call this.
-func (c *cache) ContainedIDs() util.StringSet {
-	c.lock.RLock()
-	defer c.lock.RUnlock()
-	set := util.StringSet{}
-	for id := range c.items {
-		set.Insert(id)
-	}
-	return set
-}
-
 // Get returns the requested item, or sets exists=false.
 // Get is completely threadsafe as long as you treat all items as immutable.
-func (c *cache) Get(id string) (item interface{}, exists bool) {
+func (c *cache) Get(obj interface{}) (item interface{}, exists bool, err error) {
+	id, _ := c.keyFunc(obj)
```

ignored errors should have comment with an explanation...